File: rowSumLoc.m
function [xLoc, yLoc] = rowSumLoc(dominos, line, lineLength)

[~, numDoms] = size(line);
last = line(:, numDoms);

lastDom = dominos(last(1),:);
correct = lineLength * .25;

if (last(2) == 1)   % Horizontal
    
    xLoc = lastDom(4) + 1.5 * lineLength;
    yLoc = (lastDom(1) + lastDom(2))/2 - correct;
    
else

    xLoc = lastDom(4) + .5 * lineLength;
    
    if (last(2) == 2)   % Top
        yLoc = lastDom(1) - .5 * lineLength - correct;
    end

    if (last(2) == 3)   % Bottom
        yLoc = lastDom(1) + .5 * lineLength - correct;
    end

    if (last(2) == 4)   % Middle
        yLoc = lastDom(1) - correct;
    end
end



end

File: python/domino_helper/row_sum_loc.py
"""
rowSumLoc.m -> row_sum_loc.py

Computes the (x, y) position to display the row sum text on the image.
"""


def row_sum_loc(dominos, line, line_length):
    """
    Calculate text position for a row sum label.

    Parameters
    ----------
    dominos : np.ndarray
        (num_dominos, 7) array.
    line : np.ndarray
        (2, n) array for this row.
    line_length : float

    Returns
    -------
    x_loc : float
    y_loc : float
    """
    num_doms = line.shape[1]
    last = line[:, num_doms - 1]
    last_dom = dominos[int(last[0])]
    correct = line_length * 0.25

    if last[1] == 1:  # Horizontal
        x_loc = last_dom[3] + 1.5 * line_length
        y_loc = (last_dom[0] + last_dom[1]) / 2 - correct
    else:
        x_loc = last_dom[3] + 0.5 * line_length

        if last[1] == 2:  # Top
            y_loc = last_dom[0] - 0.5 * line_length - correct
        elif last[1] == 3:  # Bottom
            y_loc = last_dom[0] + 0.5 * line_length - correct
        else:  # Middle (4)
            y_loc = last_dom[0] - correct

    return x_loc, y_loc
